api/graphql/schema.graphql

```diff
@@ -1307,6 +1308,7 @@ input CreatePoolCandidateAsAdminInput {
   status: PoolCandidateStatus = NEW_APPLICATION
     @rename(attribute: "pool_candidate_status")
   notes: String
+  submittedAt: DateTime @rename(attribute: "submitted_at")
```

Comment:
Instead of allowing the client to (optionally) set a submittedAt time, the mutation should automatically set this to the current time, with no option to override.

@petertgiles @esizer I thought we figured out a way to do this before, but I don't see any examples in the current schema. Do either of you remember anything about this?

Reply:
I think this is what you are looking for?...maybe 😅 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/198fe5a9a4b01bace0d4d0b66c7e1bf9cbe03629/api/app/GraphQL/Mutations/SubmitApplication.php#L33-L45

Reply:
I was hoping there was a way of doing it with schema directives, but that might be it.

Reply:
Maybe this SO answer could help?

https://stackoverflow.com/a/71115849

Reply:
Yeah, I was just looking at that, but I don't love that solution.

I'm thinking a good solution would be to write a custom directive, based on InjectDirective, which just gets the current time instead of getting a value from context.

Reply:
```
<?php

namespace Nuwave\Lighthouse\Schema\Directives;

use Nuwave\Lighthouse\Exceptions\DefinitionException;
use Nuwave\Lighthouse\Execution\ResolveInfo;
use Nuwave\Lighthouse\Schema\Values\FieldValue;
use Nuwave\Lighthouse\Support\Contracts\FieldMiddleware;
use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;

class InjectNowDirective extends BaseDirective implements FieldMiddleware
{
    public static function definition(): string
    {
        return /** @lang GraphQL */ <<<'GRAPHQL'
"""
Inject the current DateTime into the arguments.
"""
directive @injectNow(
  """
  The target name of the argument into which the current dateTime is injected.
  You can use dot notation to set the value at arbitrary depth
  within the incoming argument.
  """
  name: String!
) repeatable on FIELD_DEFINITION
GRAPHQL;
    }

    /**
     * @throws \Nuwave\Lighthouse\Exceptions\DefinitionException
     */
    public function handleField(FieldValue $fieldValue, \Closure $next): FieldValue
    {
        // $contextAttributeName = $this->directiveArgValue('context');
        // if (! $contextAttributeName) {
        //     throw new DefinitionException(
        //         "The `inject` directive on {$fieldValue->getParentName()} [{$fieldValue->getFieldName()}] must have a `context` argument"
        //     );
        // }

        $argumentName = $this->directiveArgValue('name');
        if (! $argumentName) {
            throw new DefinitionException(
                "The `inject` directive on {$fieldValue->getParentName()} [{$fieldValue->getFieldName()}] must have a `name` argument"
            );
        }

        $previousResolver = $fieldValue->getResolver();

        $fieldValue->setResolver(function ($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo) use ($contextAttributeName, $argumentName, $previousResolver) {
            // $valueFromContext = data_get($context, $contextAttributeName);

            // $resolveInfo->argumentSet->addValue($argumentName, $valueFromContext);

            $resolveInfo->argumentSet->addValue($argumentName, now());

            return $previousResolver(
                $rootValue,
                $resolveInfo->argumentSet->toArray(),
                $context,
                $resolveInfo
            );
        });

        return $next($fieldValue);
    }
}
```

Reply:
I removed it from the front-end in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6041/commits/60772a75f5c969a12c34e051f1f30c6bf8b19b26
and added in the above directive in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6041/commits/52adcf582d295c35df2726bfc54ff302e3290b56

I can't quite get it to actually work however, not too sure what the problem is

Reply:
Try this?
```
createPoolCandidateAsAdmin(
    poolCandidate: CreatePoolCandidateAsAdminInput! @spread
  ): PoolCandidate @create @guard @can(ability: "create") @injectNow(name: "submitted_at")
```

Reply:
oh yeah... good call

fixed in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6041/commits/bf59e0bba60a9373f6789a2aff539edaf7c78955
attempted a test in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6041/commits/0c6feeab9dc904282ae37afe470a73af4ee703da
and cleared an outstanding to do on that input in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6041/commits/effae10be212b964c745d0715e6f5597e4eef3f2